xdsl/transforms/stencil_to_csl_stencil.py

```diff
@@ -171,10 +294,282 @@ def match_and_rewrite(self, op: dmp.SwapOp, rewriter: PatternRewriter, /):
             nested_rewriter.rewrite_op(new_apply_op)
 
 
+def get_op_split(
+    ops: Sequence[Operation], buf: BlockArgument
+) -> tuple[Sequence[Operation], Sequence[Operation]]:
+    """
+    Returns a split of `ops` into an `(a,b)` tuple, such that:
+
+    - `a` contains neighbour accesses to `buf` plus the minumum set of instructions to reduce the accessed data to 1 thing
+    - `b` contains everything else
+
+    If no valid split can be found, return `(ops, [])`.
+
+    This function does not attempt to arithmetically re-structure the computation to obtain a good split. To do this,
+    `RestructureSymmetricReductionPattern()` may be executed first.
+    """
+    a: Sequence[Operation] = []
+    b: Sequence[Operation] = []
+    rem: Sequence[Operation] = []
+    for op in ops:
+        if isinstance(op, csl_stencil.AccessOp):
+            (b, a)[op.op == buf].append(op)
+        else:
+            rem.append(op)
+
+    # loop until we can make no more changes, or until only 1 thing computed in `a` is used outside of it
+    has_changes = True
+    while (
+        len(
+            # ops in `a` used outside of `a`
+            a_exports := set(
+                op
+                for op in a
+                for result in op.results
+                for use in result.uses
+                if use.operation not in a
+            )
+        )
+        > 1
+        and has_changes
+    ):
+        has_changes = False
+
+        # ops that directly depend on `a` but are not themselves in `a`
+        fontier = set(
```

Comment:
Consider more descriptive names:

```suggestion
        depending_on_a = set(
```

Reply:
`frontier` sounds like good compiler terminology no?